tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -411,19 +420,27 @@ sub save_network_device_status_logs {
 sub post_fail_hook {
     my $self = shift;
 
-    my $log_dir = "/tmp/sriov_pcipassthru";
-
     diag("Module sriov_network_card_pci_passthrough post fail hook starts.");
-    my $vm_hostnames       = script_output "virsh list --all --name | grep sles";
-    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
-    foreach (@vm_hostnames_array)
-    {
-        save_network_device_status_logs($log_dir, $_, "post_fail_hook");
+    my $log_dir = "/tmp/sriov_pcipassthru";
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        save_network_device_status_logs($log_dir, $guest, "post_fail_hook");
     }
-
     upload_virt_logs($log_dir, "network_device_status");
     $self->SUPER::post_fail_hook;
 
+    my $downloaded_xml_dir = "/tmp/download_vm_xml";
+    #restore guests for subsequent test modules
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        remove_vm($guest);
+        if (script_run("ls $downloaded_xml_dir/$guest.xml") == 0) {
+            restore_downloaded_guests($guest, $downloaded_xml_dir);
+            record_info "Guest $guest is restored.";
+        }
```

Comment:
What if   script_run("ls $downloaded_xml_dir/$guest.xml") ne 0 ?

Reply:
do nothing.  the guest is removed and not restored. ie. the guest is missing. So... I should issue a warning.

Reply:
If the following test performs test on the missing guest, I think you should stop the whole test suite as well. There is no point to continue.

Reply:
There may be guests which survive, so test should continue. I issue a warning against the failing guest.